packages/wrangler/src/dev.ts

```diff
@@ -604,456 +561,207 @@ export async function startDev(args: StartDevOptions) {
-							mtls_certificates: undefined,
-							pipelines: undefined,
-							logfwdr: undefined,
-							unsafe: undefined,
-							assets: undefined,
-						}),
-					},
-					dev: {
-						auth: async (config) => {
-							const hotkeysDisplayed = !!unregisterHotKeys;
-							let accountId = args.accountId;
-							if (!accountId) {
-								unregisterHotKeys?.();
-								accountId = await requireAuth(config);
-								if (hotkeysDisplayed) {
-									unregisterHotKeys = registerDevHotKeys(devEnv, args);
-								}
-							}
-							return {
-								accountId,
-								apiToken: requireApiToken(),
-							};
-						},
-						remote: !args.forceLocal && args.remote,
-						server: {
-							hostname: args.ip,
-							port: args.port,
-							secure:
-								args.localProtocol === undefined
-									? undefined
-									: args.localProtocol === "https",
-							httpsCertPath: args.httpsCertPath,
-							httpsKeyPath: args.httpsKeyPath,
-						},
-						inspector: {
-							port: args.inspectorPort,
-						},
-						origin: {
-							hostname: args.host ?? args.localUpstream,
-							secure:
-								args.upstreamProtocol === undefined
-									? undefined
-									: args.upstreamProtocol === "https",
-						},
-						persist: args.persistTo,
-						liveReload: args.liveReload,
-						testScheduled: args.testScheduled,
-						logLevel: args.logLevel,
-						registry: devEnv.config.latestConfig?.dev.registry,
-						bindVectorizeToProd: args.experimentalVectorizeBindToProd,
-					},
-					legacy: {
-						site: (configParam) => {
-							const legacyAssetPaths = getResolvedLegacyAssetPaths(
-								args,
-								configParam
-							);
-							return Boolean(args.site || configParam.site) && legacyAssetPaths
-								? {
-										bucket: path.join(
-											legacyAssetPaths.baseDirectory,
-											legacyAssetPaths?.assetDirectory
-										),
-										include: legacyAssetPaths.includePatterns,
-										exclude: legacyAssetPaths.excludePatterns,
-									}
-								: undefined;
-						},
-						legacyAssets: (configParam) =>
-							args.legacyAssets ?? configParam.legacy_assets,
-						enableServiceEnvironments: !(args.legacyEnv ?? true),
-					},
-					assets: args.assets,
-				} satisfies StartDevWorkerInput,
-				true
-			);
-
-			void metrics.sendMetricsEvent(
-				"run dev",
-				{
-					local: !args.remote,
-					usesTypeScript: /\.tsx?$/.test(
-						devEnv.config.latestConfig?.entrypoint as string
-					),
-				},
-				{
-					sendMetrics: devEnv.config.latestConfig?.sendMetrics,
-					offline: !args.remote,
-				}
-			);
-
-			return { devEnv, unregisterHotKeys, teardownRegistryPromise };
-		} else {
-			const projectRoot = configPath && path.dirname(configPath);
-			let config = readConfig(configPath, args);
-
 			if (
-				(args.legacyAssets || config.legacy_assets) &&
-				(args.site || config.site)
+				(args.enableIpc || !args.onReady) &&
+				process.send &&
+				typeof vitest === "undefined"
 			) {
```

Comment:
This is a change in behaviour to properly enable `unstable_dev()` with Pages. Previously, the `DEV_SERVER_READY` event was _only_ sent if `process.send()` was defined (and we weren't in tests). Now, there are two ways of reporting the port:
- If the `onReady()` callback is provided, use that
- Otherwise, if `process.send()` is defined, use that.

These two options are _usually_ mutually exclusive (if you're running `unstable_dev()` you don't usually want to send an IPC message to the parent process). However, the Pages use of `unstable_dev` relied on `DEV_SERVER_READY` to make the `workers-sdk` fixture tests run using `runWranglerLongLived()`, and so the `enableIpc` flag turns on _both_ of these port reporting mechanisms. Long-term, we should refactor Pages to use `startWorker()` and our fixture tests to not rely on IPC messages to detect the port (similar to the Wrangler E2E tests), which will allow us to remove this logic.